From 9081fca9afebe5feb27215e5896ec980bdb7330d Mon Sep 17 00:00:00 2001
From: Ralph Boehme <slow@samba.org>
Date: Thu, 23 Jun 2016 17:14:55 +0200
Subject: [PATCH 5/8] s3/smbd: move check for "hide files" to
 dos_mode_from_name()

Consolidate the "hide dot files" and "hide files" handling stuff in one
function. No change in overall behaviour.

Bug: https://bugzilla.samba.org/show_bug.cgi?id=11992

Signed-off-by: Ralph Boehme <slow@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>
(cherry picked from commit f2a53efb1aab0986d6a7d9621b1efff2127df4e6)
---
 source3/smbd/dosmode.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/source3/smbd/dosmode.c b/source3/smbd/dosmode.c
index f614729..1574166 100644
--- a/source3/smbd/dosmode.c
+++ b/source3/smbd/dosmode.c
@@ -577,6 +577,12 @@ static uint32_t dos_mode_from_name(connection_struct *conn,
 		}
 	}
 
+	if (!(result & FILE_ATTRIBUTE_HIDDEN) &&
+	    IS_HIDDEN_PATH(conn, smb_fname->base_name))
+	{
+		result |= FILE_ATTRIBUTE_HIDDEN;
+	}
+
 	return result;
 }
 
@@ -618,13 +624,6 @@ uint32_t dos_mode(connection_struct *conn, struct smb_filename *smb_fname)
 
 	result |= dos_mode_from_name(conn, smb_fname, result);
 
-	/* Optimization : Only call is_hidden_path if it's not already
-	   hidden. */
-	if (!(result & FILE_ATTRIBUTE_HIDDEN) &&
-	    IS_HIDDEN_PATH(conn, smb_fname->base_name)) {
-		result |= FILE_ATTRIBUTE_HIDDEN;
-	}
-
 	if (result == 0) {
 		result = FILE_ATTRIBUTE_NORMAL;
 	}
-- 
2.9.0

